packages/contracts/contracts/voting/allowlist/AllowlistVoting.sol

```diff
@@ -0,0 +1,235 @@
+    Checkpoints.History private _allowlistLengthCheckpoints;
+
+    /// @notice Thrown when a sender is not allowed to create a vote.
+    /// @param sender The sender address.
+    error VoteCreationForbidden(address sender);
+
+    /// @notice Emitted when new users are added to the allowlist.
+    /// @param users The array of user addresses to be added.
+    event UsersAdded(address[] users);
+
+    /// @notice Emitted when users are removed from the allowlist.
+    /// @param users The array of user addresses to be removed.
+    event UsersRemoved(address[] users);
+
+    /// @notice Initializes the component.
+    /// @dev This method is required to support [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822).
+    /// @param _dao The IDAO interface of the associated DAO.
+    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
+    /// @param _participationRequiredPct The minimal required participation in percent.
+    /// @param _supportRequiredPct The minimal required support in percent.
+    /// @param _minDuration The minimal duration of a vote.
+    /// @param _allowlisted The allowlisted addresses.
+    function initialize(
+        IDAO _dao,
+        address _trustedForwarder,
+        uint64 _participationRequiredPct,
+        uint64 _supportRequiredPct,
+        uint64 _minDuration,
+        address[] calldata _allowlisted
+    ) public initializer {
+        _registerStandard(ALLOWLIST_VOTING_INTERFACE_ID);
+        __MajorityVotingBase_init(
+            _dao,
+            _trustedForwarder,
+            _participationRequiredPct,
+            _supportRequiredPct,
+            _minDuration
+        );
+
+        // add allowlisted users
+        _addAllowlistedUsers(_allowlisted);
+    }
+
+    /// @notice Returns the version of the GSN relay recipient.
+    /// @dev Describes the version and contract for GSN compatibility.
+    function versionRecipient() external view virtual override returns (string memory) {
+        return "0.0.1+opengsn.recipient.AllowlistVoting";
+    }
+
+    /// @notice Adds new users to the allowlist.
+    /// @param _users The addresses of the users to be added.
+    function addAllowlistedUsers(address[] calldata _users)
+        external
+        auth(MODIFY_ALLOWLIST_PERMISSION_ID)
+    {
+        _addAllowlistedUsers(_users);
+    }
+
+    /// @notice Internal function to add new users to the allowlist.
+    /// @param _users The addresses of users to be added.
+    function _addAllowlistedUsers(address[] calldata _users) internal {
+        _allowlistUsers(_users, true);
+
+        emit UsersAdded(_users);
+    }
+
+    /// @notice Removes users from the allowlist.
+    /// @param _users The addresses of the users to be removed.
+    function removeAllowlistedUsers(address[] calldata _users)
+        external
+        auth(MODIFY_ALLOWLIST_PERMISSION_ID)
+    {
+        _allowlistUsers(_users, false);
```

Comment:
How could this make any sense? The function is called `remove` and the inner call is `allowlistUsers` ?? :warning:  :warning:  :warning:

Reply:
Renamed it to `_updateAllowedUsers`. Closing this conversation to discuss it in one place only (https://github.com/aragon/core/pull/62#discussion_r928945438).